subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/ConsumerProvidedResolvedVariant.java

```diff
@@ -54,6 +54,12 @@ public Completion startVisit(BuildOperationQueue<RunnableBuildOperation> actions
 
     @Override
     public void visitDependencies(TaskDependencyResolveContext context) {
-        context.add(new DefaultTransformationDependency(transformation, delegate, resolvableDependencies));
+        ExtraExecutionGraphDependenciesResolverFactory extraExecutionGraphDependenciesResolverFactory;
+        if (context instanceof TransformHackContext) {
```

Comment:
As the name imply, I am not thrilled by this way of passing `ResolverResults` down but did not find an alternate path thus far.

Reply:
It feels wrong to add a method on `TaskDependencyResolveContext` for this. Especially as there will only be that specific code path needing that info and many implementations having to implement the method ... and unable to do so except by returning an always empty factory.

Reply:
And I cannot do it without pushing `ResolverResults` and all its dependent types down into `model-core` module.